components/cdc/tests/failpoints/test_observe.rs

```diff
@@ -125,3 +133,85 @@ fn test_stale_observe_cmd() {
     event_feed_wrap.as_ref().replace(None);
     suite.stop();
 }
+
+#[test]
+fn test_delayed_change_cmd() {
+    let mut cluster = new_server_cluster(1, 3);
+    configure_for_lease_read(&mut cluster, Some(50), Some(20));
+    cluster.cfg.raft_store.raft_store_max_leader_lease = ReadableDuration::millis(100);
+    cluster.pd_client.disable_default_operator();
+    let mut suite = TestSuite::with_cluster(3, cluster);
+    suite.cluster.must_put(b"k1", b"v1");
+    let region = suite.cluster.pd_client.get_region(&[]).unwrap();
+    let leader = new_peer(1, 1);
+    suite
+        .cluster
+        .must_transfer_leader(region.get_id(), leader.clone());
+    // Wait util lease expired
+    sleep_ms(300);
+
+    let (sx, rx) = mpsc::sync_channel::<RaftMessage>(1);
+    let send_flag = Arc::new(AtomicBool::new(true));
+    let send_read_index_filter = RegionPacketFilter::new(region.get_id(), leader.get_store_id())
+        .direction(Direction::Send)
+        .msg_type(MessageType::MsgHeartbeat)
+        .set_msg_callback(Arc::new(move |msg: &RaftMessage| {
+            if send_flag.compare_and_swap(true, false, Ordering::SeqCst) {
```

Comment:
Why send heartbeat once?

Reply:
Because thread will be blocked if there is not any receiver.